packages/oak-addon-richtext-field-prosemirror/lib/core/Editor/Toolbar/index.test.js

```diff
@@ -0,0 +1,187 @@
+    const onToggleLinkMock = jest.fn();
+    const { getByText, container } = render(
+      <Toolbar
+        onToggleMark={() => jest.fn()}
+        onToggleBlock={() => jest.fn()}
+        onToggleLink={onToggleLinkMock}
+        state={{}}
+      />
+    );
+    fireEvent.click(getByText('link'));
+    fireEvent.change(
+      container.querySelector('.oak-link-url input'),
+      { target: { value: 'http://example.com' } }
+    );
+    await waitFor(() => expect(onToggleLinkMock).toHaveBeenCalledWith({
+      href: 'http://example.com',
+    }));
+  });
+
+  it('should display the right size', async () => {
+    utils.getActiveAttrs.mockReturnValue({ size: '17px' });
+
+    const { container } = render(
+      <Toolbar
+        onToggleMark={() => jest.fn()}
+        onToggleBlock={() => jest.fn()}
+        onToggleLink={() => jest.fn()}
+        state={{}}
+      />
+    );
+    await waitFor(() => expect(
+      container.querySelector('.oak-text-size').textContent
+    ).toEqual('17'));
+  });
+
+  it('should decrease size when clicking on "-" button', async () => {
+    utils.getActiveAttrs.mockReturnValue({ size: '17px' });
+    const onToggleMarkMock = jest.fn();
+    const { container, getByText } = render(
+      <Toolbar
+        onToggleMark={ onToggleMarkMock }
+        onToggleBlock={() => jest.fn()}
+        onToggleLink={() => jest.fn()}
+        state={{}}
+      />
+    );
+    await waitFor(() => expect(
+      container.querySelector('.oak-text-size').textContent
+    ).toEqual('17'));
+    fireEvent.click(getByText('horizontal_rule'));
+    expect(onToggleMarkMock)
+      .toHaveBeenCalledWith(schema.marks.size, { size: '16px' });
+  });
+
+  it('should increase size when clicking on "+" button', async () => {
+    utils.getActiveAttrs.mockReturnValue({ size: '17px' });
+    const onToggleMarkMock = jest.fn();
+    const { container, getByText } = render(
+      <Toolbar
+        onToggleMark={ onToggleMarkMock }
+        onToggleBlock={() => jest.fn()}
+        onToggleLink={() => jest.fn()}
+        state={{}}
+      />
+    );
+    await waitFor(() => expect(
+      container.querySelector('.oak-text-size').textContent
+    ).toEqual('17'));
+    fireEvent.click(getByText('add'));
+    expect(onToggleMarkMock)
+      .toHaveBeenCalledWith(schema.marks.size, { size: '18px' });
+  });
+
+  it('should set active mode on a markButton if mark is active', async () => {
+    utils.isMarkActive.mockReturnValue(true);
+
+    const { container } = render(
+      <Toolbar
+        onToggleMark={ () => jest.fn() }
+        onToggleBlock={() => jest.fn()}
+        onToggleLink={() => jest.fn()}
+        state={{}}
+      />
+    );
+    await waitFor(
+      () => expect(container.querySelector('.oak-underline.oak-active'))
+        .toBeTruthy()
+    );
+  });
+
+  it('should set active mode on a alignment ' +
+  'if alignment is active', async () => {
+    utils.getAlignment.mockReturnValue('left');
```

Comment:
mockReturnValueOnce might be more appropriate to avoid interfering with future tests

Reply:
I used mockReturnValue safely thanks to the `beforeEach` but I'll change it to a `mockReturnValueOnce` to be sure